[core] Referenced Rulesets do not emit details on validation errors

<!-- Please, prefix the report title with the language it applies to within brackets, such as [java] or [apex].
If not specific to a language, you can use [core]. -->

**Affects PMD Version:** 

7.0.0

**Description:**

If a top level rule-set XML file contains a reference to another rule-set XML file, the referenced (AKA child) file will never emit an error message when validation fails, making it impossible to troubleshoot.

Referencing the ruleset directly however will work as expected.

**Exception Stacktrace:**

```
net.sourceforge.pmd.lang.rule.RuleSetLoadException: Cannot load ruleset ./submodules/framework/pmd/rulesets/apex/unit-testing.xml: An XML validation error occurred
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:196)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRuleSetReferenceNode(RuleSetFactory.java:389)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRuleNode(RuleSetFactory.java:335)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRulesetNode(RuleSetFactory.java:242)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:186)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:248)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:139)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadRuleSetsWithoutException(RuleSetLoader.java:202)
        at net.sourceforge.pmd.lang.rule.InternalApiBridge.loadRuleSetsWithoutException(InternalApiBridge.java:40)
        at net.sourceforge.pmd.PmdAnalysis.create(PmdAnalysis.java:199)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:311)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:48)
        at net.sourceforge.pmd.cli.internal.PmdRootLogger.executeInLoggingContext(PmdRootLogger.java:55)
        at net.sourceforge.pmd.cli.commands.internal.AbstractAnalysisPmdSubcommand.execute(AbstractAnalysisPmdSubcommand.java:111)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:30)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:16)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at net.sourceforge.pmd.cli.PmdCli.main(PmdCli.java:24)

```

**Code Sample demonstrating the issue:**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ruleset name="parent"
	xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">
	<description>Parent that includes a child.</description>

	<rule ref="child.xml" />

</ruleset>
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ruleset name="child" xmlns="http://pmd.sourceforge.net/ruleset/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 https://pmd.sourceforge.io/ruleset_2_0_0.xsd">

	<description>Child.</description>

	<rule name="ApexUnitTestsMethodsShouldHaveAsserts"
		language="apex"
		message="Child message"
		class="net.sourceforge.pmd.lang.rule.XPathRule">
		<description>
			Update to the standard &quot;Apex Unit Test must have assertions&quot;
			that includes checking for Ortoo specific assertion methods
			and the use of Amoss verify methods.
		</description>
		<priority>1</priority>
		<properties>
			<property name="version" value="2.0"/>
			<property name="xpath">
				<value>
					<![CDATA[
						//Method[ModifierNode/Annotation[lower-case(@Image)='istest']]
						[
							not(
									.//MethodCallExpression[lower-case(@MethodName)=('assertequals','assertnotequals','assert')][ReferenceExpression[lower-case(@Image)='system']]
								or	.//MethodCallExpression[ReferenceExpression/lower-case(@Image)=('assert')]
								or .//MethodCallExpression[lower-case(@MethodName)='verify'])
						]
					]]>
				</value>
			</property>
		</properties>
	</rule>
</ruleset>
```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

1.  Create the above
2.  Run PMD

**Running PMD through:** *[Other]*

Ahhhhh

I have to admit, I skim read the release notes through several times thinking that there might be a breaking change, and it was seriously opaque to me.

Everything I read appeared to describe things that related specifically to 'if you are building rules using Java' - and so as a dabbler in configuring solely via xml, it looked to my eyes that there was no need to change anything *in the xml structure*.

I was aware that the XPATHs themselves may need to change - but couldn't see why that would throw an invalid XML error.

I would still struggle to pick out the pieces of the docs that relate to migrating the XML config to the new format.

The most complete logs I'm aware of retrieving... AKA --verbose.

```
"/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/bin/pmd" check --no-progress -f csv --no-cache -d "/Users/xxx/Work/xxx/pmd-ruleset.xml" -R "/Users/xxx/Work/xxx/pmd-ruleset.xml" --verbose  
[DEBUG] Log level is at TRACE
[DEBUG] Created new FileCollector with LanguageVersionDiscoverer(LanguageRegistry(apex, pom, visualforce, wsdl, xml, xsl))
[DEBUG] Adding regular file /Users/xxx/Work/xxx/pmd-ruleset.xml.
[TRACE] Adding file /Users/xxx/Work/xxx/pmd-ruleset.xml (lang: xml 1.1) 
[ERROR] Cannot load ruleset child.xml: An XML validation error occurred
[DEBUG] net.sourceforge.pmd.lang.rule.RuleSetLoadException: Cannot load ruleset child.xml: An XML validation error occurred
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:196)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRuleSetReferenceNode(RuleSetFactory.java:389)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRuleNode(RuleSetFactory.java:335)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.parseRulesetNode(RuleSetFactory.java:242)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:186)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:248)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:139)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadRuleSetsWithoutException(RuleSetLoader.java:202)
        at net.sourceforge.pmd.lang.rule.InternalApiBridge.loadRuleSetsWithoutException(InternalApiBridge.java:40)
        at net.sourceforge.pmd.PmdAnalysis.create(PmdAnalysis.java:199)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:311)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:48)
        at net.sourceforge.pmd.cli.internal.PmdRootLogger.executeInLoggingContext(PmdRootLogger.java:55)
        at net.sourceforge.pmd.cli.commands.internal.AbstractAnalysisPmdSubcommand.execute(AbstractAnalysisPmdSubcommand.java:111)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:30)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:16)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at net.sourceforge.pmd.cli.PmdCli.main(PmdCli.java:24)

[DEBUG] Runtime classpath:
/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/conf:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upack_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-system-backend-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/guava-33.0.0-jre.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-api-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/picocli-4.7.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-common-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-apex-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-extras-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-simple-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk8-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pcollections-3.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-collection-compat_2.13-2.1.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/j2objc-annotations-2.8.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-core_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-reflect-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-2.10.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pkgforce_2.13-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/asm-9.6.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-library-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/ujson_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apex-parser-3.6.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-qual-2.11.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/annotations-13.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apexlink-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/failureaccess-1.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk7-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-cli-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-implicits_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/Saxon-HE-10.9.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jline-3.21.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-core-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/progressbar-0.9.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/runforce-55.5.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/error_prone_annotations-2.23.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/summit-ast-2.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/geny_2.13-0.6.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jul-to-slf4j-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-xml-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-parallel-collections_2.13-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-xml_2.13-1.3.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/nice-xml-messages-3.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/antlr4-runtime-4.9.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-visualforce-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr305-3.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-compat-qual-2.5.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-designer-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc-upickle-json-serializer_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr250-api-1.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/commons-lang3-3.14.0.jar
[DEBUG] Aux classpath: jdk.internal.loader.ClassLoaders$AppClassLoader@7e32c033
```

Good shout.

Yep, that does give more info.

```
"/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/bin/pmd" check --no-progress -f csv --no-cache -d "/Users/xxx/Work/xxx/child.xml" -R "/Users/xxx/Work/xxx/child.xml" --verbose
[DEBUG] Log level is at TRACE
[DEBUG] Created new FileCollector with LanguageVersionDiscoverer(LanguageRegistry(apex, pom, visualforce, wsdl, xml, xsl))
[DEBUG] Adding regular file /Users/xxx/Work/xxx/child.xml.
[TRACE] Adding file /Users/xxx/Work/xxx/child.xml (lang: xml 1.1) 
[ERROR] Error at /Users/xxx/Work/xxx/child.xml:16:4
 14|            <priority>1</priority>
 15|            <properties>
 16|                    <property name="version" value="3.1"/>
        ^^^^^^^^^ Cannot set non-existent property 'version' on rule ApexUnitTestsMethodsShouldHaveAsserts

 17|                    <property name="xpath">
 18|                            <value>
[ERROR] Cannot load ruleset /Users/xxx/Work/xxx/child.xml: An XML validation error occurred
[DEBUG] net.sourceforge.pmd.lang.rule.RuleSetLoadException: Cannot load ruleset /Users/xxx/Work/xxx/child.xml: An XML validation error occurred
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.readDocument(RuleSetFactory.java:196)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:119)
        at net.sourceforge.pmd.lang.rule.RuleSetFactory.createRuleSet(RuleSetFactory.java:114)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:248)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadFromResource(RuleSetLoader.java:139)
        at net.sourceforge.pmd.lang.rule.RuleSetLoader.loadRuleSetsWithoutException(RuleSetLoader.java:202)
        at net.sourceforge.pmd.lang.rule.InternalApiBridge.loadRuleSetsWithoutException(InternalApiBridge.java:40)
        at net.sourceforge.pmd.PmdAnalysis.create(PmdAnalysis.java:199)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:311)
        at net.sourceforge.pmd.cli.commands.internal.PmdCommand.doExecute(PmdCommand.java:48)
        at net.sourceforge.pmd.cli.internal.PmdRootLogger.executeInLoggingContext(PmdRootLogger.java:55)
        at net.sourceforge.pmd.cli.commands.internal.AbstractAnalysisPmdSubcommand.execute(AbstractAnalysisPmdSubcommand.java:111)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:30)
        at net.sourceforge.pmd.cli.commands.internal.AbstractPmdSubcommand.call(AbstractPmdSubcommand.java:16)
        at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
        at picocli.CommandLine.access$1500(CommandLine.java:148)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
        at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
        at picocli.CommandLine.execute(CommandLine.java:2170)
        at net.sourceforge.pmd.cli.PmdCli.main(PmdCli.java:24)

[DEBUG] Runtime classpath:
/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/conf:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upack_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-system-backend-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/guava-33.0.0-jre.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-api-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/picocli-4.7.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-common-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-apex-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-extras-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-simple-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk8-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pcollections-3.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-collection-compat_2.13-2.1.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/j2objc-annotations-2.8.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-core_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-reflect-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-2.10.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pkgforce_2.13-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/asm-9.6.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-library-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/ujson_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apex-parser-3.6.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-qual-2.11.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/annotations-13.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apexlink-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/failureaccess-1.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk7-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-cli-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-implicits_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/Saxon-HE-10.9.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jline-3.21.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-core-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/progressbar-0.9.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/runforce-55.5.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/error_prone_annotations-2.23.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/summit-ast-2.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/geny_2.13-0.6.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jul-to-slf4j-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-xml-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-parallel-collections_2.13-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-xml_2.13-1.3.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/nice-xml-messages-3.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/antlr4-runtime-4.9.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-visualforce-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr305-3.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-compat-qual-2.5.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-designer-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc-upickle-json-serializer_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr250-api-1.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/commons-lang3-3.14.0.jar
[DEBUG] Aux classpath: jdk.internal.loader.ClassLoaders$AppClassLoader@7e32c033
```

Removing the version property means that the error disappears.

```
"/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/bin/pmd" check --no-progress -f csv --no-cache -d "/Users/xxx/Work/xxx/force-app/main/default/classes/ErrorCodes.cls" -R "/Users/xxx/Work/xxx/pmd-ruleset.xml" --verbose
[DEBUG] Log level is at TRACE
[DEBUG] Created new FileCollector with LanguageVersionDiscoverer(LanguageRegistry(apex, pom, visualforce, wsdl, xml, xsl))
[DEBUG] Adding regular file /Users/xxx/Work/xxx/force-app/main/default/classes/ErrorCodes.cls.
[TRACE] Adding file /Users/xxx/Work/xxx/force-app/main/default/classes/ErrorCodes.cls (lang: apex 60) 
[DEBUG] Rules loaded from /Users/xxx/Work/xxx/pmd-ruleset.xml:
[DEBUG] - ApexUnitTestsMethodsShouldHaveAsserts (Apex)
[DEBUG] Runtime classpath:
/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/conf:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upack_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-system-backend-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/guava-33.0.0-jre.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-api-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/picocli-4.7.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-common-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-apex-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-extras-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/slf4j-simple-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk8-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pcollections-3.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-collection-compat_2.13-2.1.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/j2objc-annotations-2.8.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-core_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-reflect-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/gson-2.10.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pkgforce_2.13-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/asm-9.6.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-library-2.13.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/ujson_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apex-parser-3.6.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-qual-2.11.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/annotations-13.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/apexlink-2.3.7.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/failureaccess-1.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/kotlin-stdlib-jdk7-1.7.20.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-cli-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/upickle-implicits_2.13-1.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/Saxon-HE-10.9.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jline-3.21.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-core-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/progressbar-0.9.5.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/runforce-55.5.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/error_prone_annotations-2.23.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/summit-ast-2.2.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/geny_2.13-0.6.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jul-to-slf4j-1.7.36.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-xml-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-parallel-collections_2.13-1.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-xml_2.13-1.3.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/nice-xml-messages-3.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/antlr4-runtime-4.9.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-visualforce-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr305-3.0.2.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/checker-compat-qual-2.5.3.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/pmd-designer-7.0.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/scala-json-rpc-upickle-json-serializer_2.13-1.0.1.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/flogger-0.7.4.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/jsr250-api-1.0.jar:/Users/xxx/.vscode/extensions/chuckjonas.apex-pmd-0.7.0/bin/pmd/lib/commons-lang3-3.14.0.jar
[DEBUG] Aux classpath: jdk.internal.loader.ClassLoaders$AppClassLoader@7e32c033
[TRACE] Using Apex version ''apex 60''
[DEBUG] MultiFile Analysis created for Optional[]
[TRACE] Processing file (lang: apex+version:60): /Users/xxx/Work/xxx/force-app/main/default/classes/ErrorCodes.cls
[DEBUG] Using rule chain for XPath rule: ApexUnitTestsMethodsShouldHaveAsserts (child)
"Problem","Package","File","Priority","Line","Description","Rule set","Rule"
```

As you say, I think it's as simple as the child XML's errors not being properly reported.

--- a/docs/pages/release_notes.md
+++ b/docs/pages/release_notes.md
@@ -16,6 +16,8 @@ This is a {{ site.pmd.release_type }} release.
 
 ### 🐛 Fixed Issues
 
+* core
+  * [#4978](https://github.com/pmd/pmd/issues/4978): \[core] Referenced Rulesets do not emit details on validation errors
 * java-bestpractices
   * [#4278](https://github.com/pmd/pmd/issues/4278): \[java] UnusedPrivateMethod FP with Junit 5 @MethodSource and default factory method name
   * [#4975](https://github.com/pmd/pmd/issues/4975): \[java] UnusedPrivateMethod false positive when using @MethodSource on a @Nested test

--- a/pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/RuleSetFactory.java
+++ b/pmd-core/src/main/java/net/sourceforge/pmd/lang/rule/RuleSetFactory.java
@@ -658,7 +658,9 @@ final class RuleSetFactory {
         return new RuleSetLoader().loadResourcesWith(resourceLoader)
                                   .filterAbovePriority(minimumPriority)
                                   .warnDeprecated(warnDeprecated)
-                                  .includeDeprecatedRuleReferences(includeDeprecatedRuleReferences);
+                                  .includeDeprecatedRuleReferences(includeDeprecatedRuleReferences)
+                                  .withReporter(reporter)
+                                  .withLanguages(languageRegistry);
     }
 
     private @NonNull XmlMessageHandler getXmlMessagePrinter() {

